debug-ui/scripts/stats-server.ts

```diff
@@ -1420,17 +1464,28 @@ const run = async () => {
   // eslint-disable-next-line no-constant-condition
   while (true) {
     try {
+      // Do checkpoint operations first
       statsServer.saveCheckpoints();
 
-      // Run depth probe and wait for next checkpoint
-      await Promise.all([
-        statsServer.depthProbe(),
-        sleep(CHECKPOINT_DURATION_SEC * 1_000),
-        DATABASE_URL ? statsServer.saveState() : () => {},
-      ]);
+      // Run depth probe
+      await statsServer.depthProbe();
+
+      // Handle database operations separately with their own error handling
+      if (DATABASE_URL) {
+        try {
+          await statsServer.saveState();
+        } catch (dbError) {
+          console.error('Database save error (continuing operation):', dbError);
+          // Allow continuing even if DB save fails
+        }
+      }
+
+      // Sleep until next checkpoint
```

Comment:
keep these batched together. if you dont, the sleep will be an hour on top of the time it takes to do the rest.

The concern with it running on the first time through gets addresses by putting a full hour sleep ahead of the while loop.